e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -667,6 +667,112 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
             }
         }
 
+        /// <summary>
+        /// Equivalent to GetSetOfUniqueNodesAsync
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="endpointId">Id of the endpoint as returned by <see cref="Registry_GetEndpoints(IIoTPlatformTestContext)"/></param>
+        /// <param name="nodeId">Id of the parent node or null to browse the root node</param>
+        /// <param name="continuationToken"></param>
+        /// <returns></returns>
+        public static async Task<List<(string NodeId, string NodeClass, bool Children)>> Twin_GetBrowseEndpoint(
+                IIoTPlatformTestContext context,
+                string endpointId,
+                string nodeId = null) {
+
+            if (string.IsNullOrEmpty(endpointId)) {
+                throw new ArgumentNullException(nameof(endpointId));
+            }
+
+            var result = new List<(string NodeId, string NodeClass, bool Children)>();
+            string continuationToken = null;
+
+            do {
+                var browseResult = await Twin_GetBrowseEndpoint_Internal(context, endpointId, nodeId, continuationToken);
+
+                if (browseResult.results.Count > 0) {
+                    result.AddRange(browseResult.results);
+                }
+
+                continuationToken = browseResult.continuationToken;
+            } while (continuationToken != null);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Equivalent to recursive calling GetSetOfUniqueNodesAsync to get the whole hierarchy of nodes
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="endpointId">Id of the endpoint as returned by <see cref="Registry_GetEndpoints(IIoTPlatformTestContext)"/></param>
+        /// <param name="nodeClass">Class of the node to filter to or null for no filtering</param>
+        /// <param name="nodeId">Id of the parent node or null to browse the root node</param>
+        public static async Task<List<(string NodeId, string NodeClass, bool Children)>> Twin_GetBrowseEndpoint_Recursive(
```

Comment:
Please add `Async` suffix to the method name. Please add a `CancellationToken` parameter and pass it down to downstream `async` methods.